### ISSUE TITLE
Functions using org-element--parse-elements possibly broken with latest Org

### ISSUE THREAD

**Issue by myshevchuk** [ndwarshuis/om.el#34]
Hi,

In Org commit `fc80d052d`, which was recently merged into the 9.5 release, the following addition was made to `org-element--parse-elements`:

```diff
(defun org-element--parse-elements 
  (beg end mode structure granularity visible-only acc)
...
+      (org-element-put-property acc :granularity granularity)
...)
```

The definition of `org-element-put-property` does not allow for `acc` to be nil:

```elisp
(defsubst org-element-put-property (element property value)
  "In ELEMENT set PROPERTY to VALUE.
Return modified element."
  (if (stringp element) (org-add-props element nil property value)
    (setcar (cdr element) (plist-put (nth 1 element) property value))
    element))
```

I haven't investigated the scope on which Org-ml has been affected, just for example `org-ml--parse-headline-subtree-at` has the following call:

```elisp
(org-element--parse-elements b e 'first-section
                                          nil nil nil nil)
```

In effect, the above call results in `(setcar nil ...)` throwing an error `(wrong-type-argument consp nil)`.

**Comment by ndwarshuis** [comment]
Thank you for pointing this out. I haven't really looked at 9.5 yet and in general I use whatever org-mode version is shipped with emacs (which for now is still 9.4.4).

I probably won't be able to look at this for another week or so. In the meantime do you want to take a stab at a PR? Based on my very brief skim of the 9.5 org-element code it looks like all the `nil`s in `org-element--parse-elements` (there's a handful in `org-ml`) will need to have some dummy container that then gets unpacked after the parsed nodes are added to it. It's a bit crude, but I suppose this is what happens when I use a private function as the foundation for a library. :upside_down_face:

**Comment by myshevchuk** [comment]
I was mistaken to say this commit was merged into 9.5. It's not there yet, just in the main branch, so there is probably enough time before it is released. 

It'd be interesting for me to prepare a PR on this, but it will likely take some time.

**Comment by myshevchuk** [comment]
Hi, I've made a PR on this. It works for me on Org master and passes the tests on Org 9.4-9.5.
